plenum/server/replica.py

```diff
@@ -1627,3 +1628,28 @@ def send(self, msg, stat=None) -> None:
         if stat:
             self.stats.inc(stat)
         self.outBox.append(msg)
+
+    def caught_up_till_pp_seq_no(self, last_caught_up_pp_seq_no):
+        self.addToOrdered(self.viewNo, last_caught_up_pp_seq_no)
+        # self._remove_till_caught_up_pp_seq_no(last_caught_up_pp_seq_no)
+
+    def _remove_till_caught_up_pp_seq_no(self, last_caught_up_pp_seq_no):
```

Comment:
_remove_till_caught_up_pp_seq_no is not finalized yet (it's a fix for INDY-70 which is under rdevelopment). It's not called yet.